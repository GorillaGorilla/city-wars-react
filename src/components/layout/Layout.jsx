import React from 'react';
import Area from '../area/Area';
import './Layout.css';
import MapArea from '../map-area/MapArea';

const Layout = () => (
  <div className="layout">
    <Area label="header" />
    <Area label="sidebar" />
    <MapArea />
    <Area label="panel" />
  </div>
);

export default Layout;
