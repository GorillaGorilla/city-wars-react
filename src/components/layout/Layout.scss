.layout {
    grid-gap: 10px;
    height: 100vh;
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 50px 1fr 50px;
    grid-template-areas: 
        "header"
        "map"
        "panel";

    .area {
        &.sidebar {
            display: none;
        }
        &.bonus {
            display: none;
        }
    } 
}

// medium grid template
@media (min-width: 768px) {
    .layout {
        grid-template-columns: 300px 5fr 3fr;
        grid-template-areas:
            "header     header      header"
            "panel    map    map"
            "panel    map      map";

        .area {
            &.sidebar {
                display: none;
            }
        } 
    }
}
