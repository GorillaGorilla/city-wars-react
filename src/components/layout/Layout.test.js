import React from 'react';
import { shallow } from 'enzyme';
import Layout from './Layout';

describe('Layout', () => {
  const layout = shallow(<Layout />);

  it('should render without crashing', () => {
    expect(layout.exists()).toBe(true);
  });

  it('should match snapshot', () => {
    expect(layout).toMatchSnapshot();
  });

  it('should contain 4 areas', () => {
    expect(layout.children().length).toEqual(3);
  });

  // it('should contain a map area component', () => {
  //   expect(layout.find('Connect(MapContainer)').exists()).toBe(true);
  // });
});

