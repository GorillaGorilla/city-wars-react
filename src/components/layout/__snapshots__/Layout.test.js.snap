// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Layout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Area
          label="header"
/>,
        <Area
          label="sidebar"
/>,
        <MapArea />,
        <Area
          label="panel"
/>,
      ],
      "className": "layout",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "header",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "sidebar",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "panel",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Area
            label="header"
/>,
          <Area
            label="sidebar"
/>,
          <MapArea />,
          <Area
            label="panel"
/>,
        ],
        "className": "layout",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "header",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "sidebar",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "panel",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
